qjsc.c

```diff
@@ -240,12 +238,9 @@ JSModuleDef *jsc_module_loader(JSContext *ctx,
         /* create a dummy module */
         m = JS_NewCModule(ctx, module_name, js_module_dummy_init);
     } else if (has_suffix(module_name, ".so")) {
-        fprintf(stderr, "Warning: binary module '%s' will be dynamically loaded\n", module_name);
-        /* create a dummy module */
-        m = JS_NewCModule(ctx, module_name, js_module_dummy_init);
-        /* the resulting executable will export its symbols for the
-           dynamic library */
-        dynamic_export = TRUE;
+        JS_ThrowReferenceError(ctx, "could not load module '%s' it's a shared library module",
```

Comment:
Maybe rephrase with a little additional detail?
```suggestion
        JS_ThrowReferenceError(ctx, "%s: dynamically linking to shared libraries not supported",
```